pkg/controller/import-controller.go

```diff
@@ -792,6 +820,127 @@ func createImporterPod(log logr.Logger, client client.Client, image, verbose, pu
 	return pod, nil
 }
 
+// makeCriImporterPodSpec creates and return the container runtime interface importer pod spec based on the passed-in importImage and pvc.
+func makeCriImporterPodSpec(namespace, image, importImage, pullPolicy string, pvc *corev1.PersistentVolumeClaim, podResourceRequirements *corev1.ResourceRequirements, workloadNodePlacement *sdkapi.NodePlacement, priorityClassName string) *corev1.Pod {
+	// importer pod name contains the pvc name
+	podName, _ := pvc.Annotations[AnnImportPod]
+
+	blockOwnerDeletion := true
+	isController := true
+
+	volumes := []corev1.Volume{
+		{
+			Name: "shared-volume",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: DataVolName,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: pvc.Name,
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				AnnCreatedBy: "yes",
+			},
+			Labels: map[string]string{
+				common.CDILabelKey:       common.CDILabelValue,
+				common.CDIComponentLabel: common.ImporterPodName,
+				common.PrometheusLabel:   "",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         "v1",
+					Kind:               "PersistentVolumeClaim",
+					Name:               pvc.Name,
+					UID:                pvc.GetUID(),
+					BlockOwnerDeletion: &blockOwnerDeletion,
+					Controller:         &isController,
+				},
+			},
+		},
+		Spec: corev1.PodSpec{
+			InitContainers: []corev1.Container{
+				{
+					Name:    "init",
+					Image:   image,
+					Command: []string{"sh", "-c", "cp /usr/bin/cdi-containerimage-server /shared/server"},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/shared",
+							Name:      "shared-volume",
+						},
+					},
+					ImagePullPolicy: corev1.PullPolicy(pullPolicy),
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:    "server",
+					Image:   importImage,
+					Command: []string{"/shared/server", "-p", "8100", "-image-dir", "/disk", "-ready-file", "/shared/ready", "-done-file", "/shared/done"},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/shared",
+							Name:      "shared-volume",
+						},
+					},
+					ImagePullPolicy: corev1.PullPolicy(pullPolicy),
+				},
+				{
+					Name:    common.ImporterPodName,
+					Image:   image,
+					Command: []string{"/usr/bin/import.sh", "8100", "/shared/ready", "/shared/done", "disk.img", "/data/disk.img"},
```

Comment:
should configure standard http import container here

Reply:
may be nicer, your call

Reply:
Might be nice to take how we create the container for a standard importer, and put that into a function, and call that function from both places?

Reply:
Done